src/Popover.js

```diff
@@ -1,249 +1,47 @@
-    }
-  }
-
-  // Top & Bottom: Left-oriented carets
-  ${Popover}.caret-pos--top-left & ,
-  ${Popover}.caret-pos--bottom-left & {
-    left: -9px;
-    margin-left: 0;
-
-    &::before,
-    &::after {
-      left: ${get('space.4')};
-      margin-left: 0;
-    }
-
-    &::after {
-      left: calc(${get('space.4')} + 1px);
-    }
-  }
-
-  // Right- & Left-oriented carets
-  ${Popover}.caret-pos--right & ,
-  ${Popover}.caret-pos--right-top & ,
-  ${Popover}.caret-pos--right-bottom & ,
-  ${Popover}.caret-pos--left & ,
-  ${Popover}.caret-pos--left-top & ,
-  ${Popover}.caret-pos--left-bottom & {
-    &::before,
-    &::after {
-      top: 50%;
-      left: auto;
-      margin-left: 0;
-      border-bottom-color: transparent;
-    }
-
-    &::before {
-      // stylelint-disable-next-line primer/spacing
-      margin-top: calc((${get('space.2')} + 1px) * -1);
-    }
-
-    &::after {
-      margin-top: -${get('space.2')};
-    }
-  }
-
-  // Right-oriented carets
-  ${Popover}.caret-pos--right & ,
-  ${Popover}.caret-pos--right-top & ,
-  ${Popover}.caret-pos--right-bottom & {
-    &::before {
-      right: -${get('space.3')};
-      border-left-color: ${get('popovers.colors.caret')};
-    }
-
-    &::after {
-      right: -14px;
-      // stylelint-disable-next-line primer/borders
-      border-left-color: ${get('colors.white')};
-    }
-  }
-
-  // Left-oriented carets
-  ${Popover}.caret-pos--left & ,
-  ${Popover}.caret-pos--left-top & ,
-  ${Popover}.caret-pos--left-bottom & {
-    &::before {
-      left: -${get('space.3')};
-      border-right-color: ${get('popovers.colors.caret')};
-    }
-
-    &::after {
-      left: -14px;
-      // stylelint-disable-next-line primer/borders
-      border-right-color: ${get('colors.white')};
-    }
-  }
-
-  // Right & Left: Top-oriented carets
-  ${Popover}.caret-pos--right-top & ,
-  ${Popover}.caret-pos--left-top & {
-    &::before,
-    &::after {
-      top: ${get('space.4')};
-    }
-  }
-
-  // Right & Left: Bottom-oriented carets
-  ${Popover}.caret-pos--right-bottom & ,
-  ${Popover}.caret-pos--left-bottom & {
-    &::before,
-    &::after {
-      top: auto;
-    }
-
-    &::before {
-      bottom: ${get('space.3')};
-    }
-
-    &::after {
-      bottom: calc(${get('space.3')} + 1px);
-    }
-  }
-
-  ${sx};
-`
-
-Popover.CARET_POSITIONS = [
-  'top',
-  'bottom',
-  'left',
-  'right',
-  'bottom-left',
-  'bottom-right',
-  'top-left',
-  'top-right',
-  'left-bottom',
-  'left-top',
-  'right-bottom',
-  'right-top'
-]
+const Popover = (props) => {
+  // don't destructure these, just grab them
+  const {bg, borderColor, borderWidth, borderStyle, relative, open} = props
+  const {caret, children, ...boxProps} = props
+  const caretProps = {bg, borderColor, borderWidth, borderStyle, location: caret}
+  return (
+    <BorderBox sx={{position: relative ? 'relative' : 'absolute'}} zIndex="100" ml="auto" mr="auto" p={4} width="232px" display={open ? 'block' : 'none'} {...boxProps}>
```

Comment:
@BinaryMuse I copied over the fixed `PointerBox` code and moved the `relative` positioning to the `BorderBox` so we 1) fix some caret positioning issues 2) don't need a `Popover` and `Popover.Content` and can just have one `Popover` component instead. Can you think of any reason why we need to have a separate wrapping container? Otherwise I think this breaking change would be worth it since it simplified the internal code & external API a bit.